x/dualstaking/keeper/grpc_query_delegator_providers_test.go

```diff
@@ -0,0 +1,120 @@
+package keeper_test
+
+import (
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/lavanet/lava/testutil/common"
+	"github.com/lavanet/lava/x/dualstaking/types"
+	epochstoragetypes "github.com/lavanet/lava/x/epochstorage/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDelegatorProvidersQuery(t *testing.T) {
+	ts := newTester(t)
+
+	ts.setupForDelegation(1, 1, 0, 0) // 1 delegator, 1 staked provider
+
+	_, delegator := ts.GetAccount(common.CONSUMER, 0)
+	_, provider := ts.GetAccount(common.PROVIDER, 0)
+
+	spec := ts.Spec("mock")
+
+	amountUint64 := uint64(100)
+	amount := sdk.NewCoin(epochstoragetypes.TokenDenom, sdk.NewIntFromUint64(amountUint64))
+
+	// delegate and query
+	_, err := ts.TxDualstakingDelegate(delegator, provider, spec.Index, amount)
+	require.Nil(t, err)
+	ts.AdvanceEpoch()
```

Comment:
with this, the test does not cover the case of query with "pending=false" before the change takes effect.
I think it's better to do both pending cases (false, true) here, like:

    delegate
    query without pending (should not show)
    query with pending (should show)
    advance epoch
    query without pending (should show)
    query with pending (should show)

and repeat that after each change to delegation. then you can drop the new test function.
(maybe refactor a helper for queries that will do both).

Reply:
I updated the WithPending test. see https://github.com/lavanet/lava/commit/4564b1721989d3de340f742ac8e86c0e7a93e348